instrumentation/rxjava-2.0/javaagent/src/main/java/io/opentelemetry/instrumentation/rxjava2/RxJava2InstrumentationModule.java

```diff
@@ -41,15 +43,23 @@ public RxJava2InstrumentationModule() {
     @Override
     public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
       return Collections.singletonMap(
-          isTypeInitializer(),
-          RxJava2InstrumentationModule.class.getName() + "$RxJavaPluginsAdvice");
+          isMethod(), RxJava2InstrumentationModule.class.getName() + "$RxJavaPluginsAdvice");
     }
   }
 
   public static class RxJavaPluginsAdvice {
+    @Advice.OnMethodEnter
+    public static int trackCallDepth() {
+      return CallDepthThreadLocalMap.incrementCallDepth(RxJavaPlugins.class);
+    }
+
     @Advice.OnMethodExit(suppress = Throwable.class)
-    public static void postStaticInitializer() {
+    public static void postStaticInitializer(@Advice.Enter int callDepth) {
+      if (callDepth > 0) {
+        return;
+      }
       TracingAssembly.enable();
+      CallDepthThreadLocalMap.reset(RxJavaPlugins.class);
```

Comment:
I just pushed a change to remove the call depth tracking in favor of a boolean so `enable()` is only called once ever - mainly since `enable` uses `synchronized` as we didn't expect the contention to matter.